### PR TITLE
Fail builds without menu

### PR DESCRIPTION
**Pull request by reinhrst**
OK, so I came up with a small plan to at least make sure we don't put a site live without a menu.

Especially since in the multi-language version, the two menus seem to be independent; so the chance that you will put a website live with 2 working menus is now squared (e.g. if the chance was 0.8 in the Polish-only, it's not 0.64 -- and I think 0.8 is generous).

Please read through the individual commits (and not the squashed version below) to understand what is going on.

Basically we fail the build when there is no menu.
Then on netlify we (hopefully) retry up to 10 times to get a working build.

(and yes, it certainly happens that a build fails and a next `hugo` execution in a loop, only milliseconds later, same code, succeeds).

### CODE REVIEW

File: layouts/partials/header.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<!-- Mark up for Site Navigation Section-->
```
Comment (DoomHammer) [inline layouts/partials/header.html:1]: Why do we need this for?

Comment (reinhrst) [reply layouts/partials/header.html:1]: I need to put the fail-script somewhere, and in the file that generates the menu seems the most logical. This  is copied from the theme.

We need changes in this file for the multi-lang stuff anyways, so not really a problem to include it now (rather than put the check somewhere illogical and then moving it when the multi-lang stuff comes around.

Also I prefer to go through merge-hell when I have to rebase the multi-lang stuff :)

Comment (reinhrst) [reply layouts/partials/header.html:1]: Best to read one commit at a time

Comment (reinhrst) [reply layouts/partials/header.html:1]: Even though it should be "Copy header from theme" not "from layouts". I'll fix this....

**Review by DoomHammer — APPROVED** [review]

**Review by DoomHammer — APPROVED** [review]
